tools/foreman-cockpit-session

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env ruby
+
+require "json"
+require "net/http"
+require "net/ssh"
+
+def dbg(str)
+  # $stderr.write(str + "\n")
+end
+
+# Cockpit protocol, encoding and decoding of control messages.
+
+def send_control(msg)
+  text = JSON.dump(msg)
+  dbg("-> #{text}")
+  $stdout.write("#{text.length+1}\n\n#{text}")
+  $stdout.flush
+end
+
+def read_size(io)
+  size = 0
+  seen = 0
+
+  while seen < 8
+    t = io.readpartial(1)
+
+    unless t
+      return 0
+    end
```

Comment:
`return 0 unless t` would be more ruby-like. This pattern repeats several times in this file

Reply:
I have changed a bunch of them,